lib/utils/patterns.ts

```diff
@@ -405,7 +405,7 @@ export async function getOnlineTargetUuid(
 			})),
 		});
 	} catch (err) {
-		if (!(err instanceof BalenaApplicationNotFound)) {
+		if (!instanceOf(err, BalenaApplicationNotFound)) {
```

Comment:
* `balena ssh <uuid>` now works, even with `balena-sdk` v12.33.0.
* Another possible version of the fix would have been:
```
if (err.name !== 'BalenaApplicationNotFound') {
```
It would even had allowed removing the `balena-errors` dependency from `package.json`. Very tempting. On balance my choice was to keep the dependency and instanceof, because relying solely on string comparison introduces its own fragility: it's prone to typo errors, the compiler does not complain if one writes `if (err.name !== 'foo_blah')` even if `'foo_blah'` is nonsense.